app/models/sufia/permission_template.rb

```diff
@@ -26,19 +26,45 @@ def release_no_delay?
       release_period == RELEASE_TEXT_VALUE_NO_DELAY
     end
 
-    # Does this permission template require a date that all works are released before
+    # Does this permission template require a date (or embargo) that all works are released before
     # NOTE: date will be in release_date
     def release_before_date?
-      release_period == RELEASE_TEXT_VALUE_BEFORE_DATE
+      # All PermissionTemplate embargoes are dynamically determined release before dates
+      release_period == RELEASE_TEXT_VALUE_BEFORE_DATE || release_max_embargo?
     end
 
-    # Is a specific embargo period required by this permission template
-    # NOTE: embargo period will be in release_period
-    def release_embargo?
+    # Is there a maximum embargo period specified by this permission template
+    # NOTE: latest embargo date returned by release_date, maximum embargo period will be in release_period
+    def release_max_embargo?
```

Comment:
There are two points of knowledge regarding the `RELEASE_TEXT_VALUE_*` constants.

What if we created the following:

```ruby
# Key/Value pair, where values are number of months (alternate they could be lambdas that are called)
RELEASE_TEXT_VALUES = {
  RELEASE_TEXT_VALUE_6_MONTHS => 6,
  RELEASE_TEXT_VALUE_1_YEAR => 12,
  RELEASE_TEXT_VALUE_2_YEARS => 24,
  RELEASE_TEXT_VALUE_3_YEARS => 36
}

def release_max_embargo?
  RELEASE_TEXT_VALUES.key?(release_period)
end

def release_date
  # If this isn't an embargo, just return release_date from database
  return self[:release_date] unless release_max_embargo?
  Time.zone.today + RELEASE_TEXT_VALUES.fetch(release_period).months
end
```